src/dnx.host/NativeMethods.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Framework.Runtime
+{
+    internal static class NativeMethods
+    {
+        public unsafe static string Uname()
+        {
+            try
+            {
+                var buffer = new byte[8192];
+                try
+                {
+                    fixed (byte* buf = buffer)
+                    {
+                        if (uname((IntPtr)buf) == 0)
+                        {
+                            return Marshal.PtrToStringAnsi((IntPtr)buf);
+                        }
+                    }
+                    return string.Empty;
+                }
+                catch
+                {
+                    return string.Empty;
+                }
+
+            }
+            catch (DllNotFoundException)
+            {
+                return null;
+            }
+        }
+
+
+        // Linux and Mac import
+        [DllImport("libc")]
+        private static extern int uname(IntPtr buf);
```

Comment:
[In my big Kestrel PR](https://github.com/Tragetaschen/KestrelHttpServer/blob/pinvoke-cleanup/src/Microsoft.AspNet.Server.Kestrel/Networking/Libuv.cs#L26), I'm using a variant that omits unsafe entirely.
[CoreFX](https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/Interop.PlatformDetection.cs#L58) uses `stackalloc`